google-cloud-datastore/src/main/java/com/google/cloud/datastore/DatastoreOptions.java

```diff
@@ -84,20 +94,50 @@ public static class Builder extends ServiceOptions.Builder<Datastore, DatastoreO
 
     private String namespace;
     private String databaseId;
+    private TransportChannelProvider channelProvider = null;
+    private CredentialsProvider credentialsProvider = null;
 
     private Builder() {}
 
     private Builder(DatastoreOptions options) {
       super(options);
-      namespace = options.namespace;
-      databaseId = options.databaseId;
+      this.namespace = options.namespace;
+      this.databaseId = options.databaseId;
+      this.channelProvider = options.channelProvider;
```

Comment:
Why aren't you using the `setChannelProvider()` here? It has additional checks.

Reply:
`setChannelProvider` returns a Builder, but I can extract out the validation and use it in both places.

Reply:
added!